### ISSUE TITLE
Support providing custom branch name for `build:env:push`

### ISSUE THREAD

**Issue by G-Rath** [pantheon-systems/terminus-build-tools-plugin#420]
Currently `build:env:push` is hard-coded to use the `master` branch if it's not a multi-dev call:

https://github.com/pantheon-systems/terminus-build-tools-plugin/blob/2426c2472f286d19ff2bb711ae13b718129859ee/src/Commands/BuildToolsBase.php#L804

We've both moved to using `main` for our branches instead of `master`, and actually want to have a specific `deployment` branch that CI commits to so that devs don't have to do asset compiling themselves (e.g. we ignore our compiled assets, and then CI builds + forcefully commits them to the `deployment` branch, which is then deployed from).

I would say long-term it would probably make sense to make a global `--default-branch` config value that gets used in these situations.

**Comment by greg-1-anderson** [comment]
Right now, the build tools plugin assumes that the GitHub (/ GitLab / Bitbucket) branch always matches the Pantheon branch. The Pantheon branch for the dev environment *must* be "master". We've gotten a lot of requests to change this, but it's hardcoded in too many places to be changed easily, so it will be a while before this can be changed on the platform.

Build tools could, however, be modified to push from the default branch on GitHub (et. al.) to the master branch on Pantheon. This was discussed once; I don't recall what the challenges were.

**Comment by G-Rath** [comment]
> Build tools could, however, be modified to push from the default branch on GitHub (et. al.) to the master branch on Pantheon.

Yeah that's what I had in mind - we're not fussed with what branch pantheon uses in the remote.

(tbh after doing more toying with this, I suspect I'm actually after a `build:env:push-commit` commit, but that can be another issue)
